src/main/java/pl/edu/agh/firecell/storage/FileSystemStorage.java

```diff
@@ -0,0 +1,81 @@
+package pl.edu.agh.firecell.storage;
+
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import io.reactivex.rxjava3.subjects.PublishSubject;
+import io.reactivex.rxjava3.subjects.Subject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.edu.agh.firecell.model.State;
+import pl.edu.agh.firecell.storage.serialization.StateSerializer;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+public class FileSystemStorage implements Storage {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final Subject<StateHolder> stateSubject = PublishSubject.create();
+    private final StateSerializer serializer;
+    private final Path path;
+
+    private Disposable subscription;
+
+    public FileSystemStorage(StateSerializer serializer, Path path) {
```

Comment:
If we pass `path` as the dependency here we will have to change it every time we run program.
I would leave it for some cases, but another constructor with default path would allow us to use the same one.
Ofc we should then clear this directory before we'll fill it with files.

Reply:
added second constructor